packages/vitest/src/node/reporters/benchmark/table/index.ts

```diff
@@ -34,6 +50,17 @@ export class TableReporter extends BaseReporter {
     await this.stopListRender()
     this.ctx.logger.log()
     await super.onFinished(files, errors)
+
+    // write output for future comparison
+    let outputFile = getOutputFile(this.ctx.config.benchmark, 'default')
+    if (outputFile) {
```

Comment:
I don't like having it here. I think we should just take data from a json report.

Reply:
What do you mean by "take data from a json report"?

This feature assumes users only use `default` bench reporter (i.e. zero config) and then add `--outputFile` or `--compare` manually from CLI, so handling output here seems appropriate.

Reply:
> To me, it doesn't make sense why running `vitest bench --outputFile=file.js` outputs a compare file.

Yeah, I get this might feel odd (I think I just got used to it while running main commands). But isn't it only about `--outputFile` naming? I feel it's alright if it's named like `--save-compare file.json`.

I think being able to iterate with just a single command `vitest bench` is important for the usability. For example, something like this workflow seems intuitive.

```sh
git checkout branch1
vitest bench --save-compare branch1.json  # this should show a usual bench output while saving a file

git checkout branch2
vitest bench --save-compare branch2.json 
vitest bench --compare branch1.json

git checkout main
vitest bench --compare branch1.json
vitest bench --compare branch2.json
```

Showing a usual bench output while saving a file seems like a nice feature since bench tends to take some time and having no terminal output makes me feel nervous.

I guess it's still possible to implement the same experience by tweaking default config (for example, if `--save-compare` is given, enable `default` and `json` reporters, etc...), but wouldn't it make implementations more convoluted?

Reply:
Ok, I agree that adding a separate config option would be better (`--save-compare` in your example) - not sure about the name.

Reply:
I don't know what was the exact motivation and the design of current json reporter (which exists since the introduction of benchmark feature https://github.com/vitest-dev/vitest/pull/1029).

For my implementation, I didn't want to bother with existing format so I've thrown away, but if having something more readable json output is still desirable, then I can rework the format together.

Reply:
Reworking the format would be great. The benchmark feature is still experimental, so we can introduce any breaking changes here.

Reply:
I'm still not sure about what's the design goal, but I restructured the compare output to be something like this https://github.com/vitest-dev/vitest/pull/5398/commits/86088a86076f0ea52df64bbf74018434e484ba2c

Actually it's mostly same but the differences are:
- remove `samples: number[]` but add `sampleCount: number` to avoid gigantic json
- add `id: number` to match easily for next run with `--compare` 
- `{ [ suiteName: string ]: Result[] }` to `{ suiteFullName: string, benchmarks: Result[] }[]`

The example output looks like this:

```sh
pnpm -C test/benchmark test -- bench --root fixtures/basic --outputFile bench.json
```

[bench.json](https://github.com/vitest-dev/vitest/files/14953125/bench.json)

If we are going to output a file from `default` reporter with `--save-compare`, then do we still need a dedicated json reporter? Of course, we can just share the same utility `createFormattedBenchamrkReport` for both reporters, but I feel it's redundant and confusing for users, so can we remove json reporter for now?

Reply:
> If we are going to output a file from `default` reporter with `--save-compare`, then do we still need a dedicated json reporter?

Yes, this is what I am afraid of. If we are going to remove the json reporter, then the flag to generate it cannot be `--save-compare` because it assumes the usage of the file. Maybe `--output-json`? I'd also like the report to have a `file`/`filepath` property somewhere.

Reply:
> Yes, this is what I am afraid of. If we are going to remove the json reporter, then the flag to generate it cannot be `--save-compare` because it assumes the usage of the file. Maybe `--output-json`?

Yeah, the flag name should be adjusted, but I don't see the purpose of json reporter, so I think we should remove it if we can.
Either `--output-json` or even current standard `--outputFile` sounds okay to me.

> I'd also like the report to have a `file`/`filepath` property somewhere.

I can tweak this one easily and `filepath` makes sense. 

<details><summary>Show example</summary>

```ts
{
  "files": [
    {
      "filepath": "/home/hiroshi/code/others/vitest/test/benchmark/fixtures/basic/base.bench.ts",
      "groups": [
        {
          "fullName": "base.bench.ts > sort",
          "benchmarks": [
            {
              "id": "-2066038406_0_0",
              "sampleCount": 5,
              "name": "normal",
...
            },
            {
              "id": "-2066038406_0_1",
              "sampleCount": 5,
              "name": "reverse",
...
            }
          ]
        },
        {
          "fullName": "base.bench.ts > timeout",
          "benchmarks": [
            {
              "id": "-2066038406_1_0",
              "sampleCount": 3,
              "name": "timeout100",
...
    {
      "filepath": "/home/hiroshi/code/others/vitest/test/benchmark/fixtures/basic/only.bench.ts",
      "groups": [
        {
          "fullName": "only.bench.ts",
          "benchmarks": [
            {
              "id": "151219967_6",
              "sampleCount": 1,
              "name": "visited",
              "rank": 1,
              "rme": 0,
```

</details>

Reply:
@sheremet-va Do you think reusing `--outputFile` is a bad idea?

Reply:
I don't like the `outputFile` because why would it output json in a default reporter? Why not the table? Or a graph?

Reply:
Ah, right. This bench reporter is doing many things now and json output is just a part of the feature.

I'm good with `--output-json`, but do you want to explore other ideas to make json reporter somehow co-exists with default reporter? Or can I remove json reporter?

Reply:
I think we can remove the json reporter.